pkg/gui/controllers/local_commits_controller.go

```diff
@@ -421,7 +421,12 @@ func (self *LocalCommitsController) amendTo(commit *models.Commit) error {
 		HandleConfirm: func() error {
 			return self.c.WithWaitingStatus(self.c.Tr.AmendingStatus, func() error {
 				self.c.LogAction(self.c.Tr.Actions.AmendCommit)
-				err := self.git.Rebase.AmendTo(commit.Sha)
+				var err error
+				if index := self.context().GetSelectedLineIdx(); index == 0 {
+					err = self.git.Commit.AmendHead()
+				} else {
+					err = self.git.Rebase.AmendTo(commit.Sha)
+				}
 				return self.helpers.MergeAndRebase.CheckMergeOrRebase(err)
```

Comment:
`self.helpers.MergeAndRebase.CheckMergeOrRebase(err)` only needs to be called after the AmendTo call, because that's the one that does the rebase

Reply:
On second thought, I'd rather just invoke `func (self *FilesController) handleAmendCommitPress` in case of it being `HEAD`, since it also handles GPG, but it's not visible from the `LocalCommitsController` struct?

Reply:
Should I chuck _it_ in a helper and invoke it from both `FilesController` and `LocalCommitsController`?

Reply:
good question. Yeah I'd chuck it in a helper. Weird to have gpg supported in one context but not the other but better than gpg not being supported in any context